std/hash_map.zig

```diff
@@ -518,123 +522,201 @@ pub fn getTrivialEqlFn(comptime K: type) (fn (K, K) bool) {
 pub fn getAutoHashFn(comptime K: type) (fn (K) u32) {
     return struct {
         fn hash(key: K) u32 {
-            comptime var rng = comptime std.rand.DefaultPrng.init(0);
-            return autoHash(key, &rng.random, u32);
+            const h = autoHash(key, 0);
+            return @truncate(u32, h);
         }
     }.hash;
 }
 
 pub fn getAutoEqlFn(comptime K: type) (fn (K, K) bool) {
     return struct {
         fn eql(a: K, b: K) bool {
-            return autoEql(a, b);
+            return meta.eql(a, b);
         }
     }.eql;
 }
 
-// TODO improve these hash functions
-pub fn autoHash(key: var, comptime rng: *std.rand.Random, comptime HashInt: type) HashInt {
-    switch (@typeInfo(@typeOf(key))) {
+/// Provides generic hashing for any eligible type.
+/// Only hashes `key` itself, pointers are not followed.
+/// The underlying hashing algorithm is wyhash.
+pub fn autoHash(key: var, seed: u64) u64 {
+    // We use the fact that wyhash takes an input seed to "chain" hasing when the
```

Comment:
@Sahnvour is this a typo?
`to "chain" hashing`

Reply:
Might not be the best wording, but that's not a typo. This is referencing the implementation of arrays or structs, for example.